### PR TITLE
[dmx] Refactor and improve code

### PR DESCRIPTION
**Pull request by J-N-K**
Also adds null-annotations

Originally contributed as https://github.com/smarthomej/addons/pull/58

Signed-off-by: Jan N. Klug <github@klug.nrw>

### CODE REVIEW

File: bundles/org.openhab.binding.dmx/src/main/java/org/openhab/binding/dmx/internal/DmxHandlerFactory.java [hunk 1]
```diff
@@ -36,15 +42,22 @@
...
  */
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.dmx")
+@NonNullByDefault
 public class DmxHandlerFactory extends BaseThingHandlerFactory {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Stream
+            .of(ArtnetBridgeHandler.SUPPORTED_THING_TYPES, Lib485BridgeHandler.SUPPORTED_THING_TYPES,
+                    SacnBridgeHandler.SUPPORTED_THING_TYPES, ChaserThingHandler.SUPPORTED_THING_TYPES,
+                    ColorThingHandler.SUPPORTED_THING_TYPES, DimmerThingHandler.SUPPORTED_THING_TYPES,
+                    TunableWhiteThingHandler.SUPPORTED_THING_TYPES)
+            .flatMap(Set::stream).collect(Collectors.toUnmodifiableSet());
```
Comment (jlaur) [inline bundles/org.openhab.binding.dmx/src/main/java/org/openhab/binding/dmx/internal/DmxHandlerFactory.java:52]: Consider `Set.of()`.

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the improvements! Only one very minor comment.

File: bundles/org.openhab.binding.dmx/src/main/java/org/openhab/binding/dmx/internal/DmxHandlerFactory.java (hunk 1 above)
Comment (J-N-K) [reply bundles/org.openhab.binding.dmx/src/main/java/org/openhab/binding/dmx/internal/DmxHandlerFactory.java:52]: That's not possible, we'll end up with a `Set<Set<ThingTypeUID>>` then.

Comment (jlaur) [reply bundles/org.openhab.binding.dmx/src/main/java/org/openhab/binding/dmx/internal/DmxHandlerFactory.java:52]: Oh, you are right, was reading too fast.

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**